packages/freezed/lib/src/templates/parameter_template.dart

```diff
@@ -157,7 +157,11 @@ class ParametersTemplate {
 
   @override
   String toString() {
-    final buffer = StringBuffer()..writeAll(requiredPositionalParameters, ', ');
+    final buffer = StringBuffer()
+      ..writeAll(
+          requiredPositionalParameters
+              .map<String>((p) => p.toString(skipRequired: true)),
```

Comment:
Instead of `p.toString(skipRequired: true)`, I think it would be more logical to have `p.copyWith(isRequired: false)`

Reply:
Thanks for taking a look - I was a bit hesitant, since it still is required, it's just positional instead of named.. do you still prefer `isRequired` vs. like `showIsRequired`?  It's very similar to the `showDefaultValue` flag? 🤷‍♂️️ - and is t he `asExpanded` method meant as a way to `copyWith` and can it be reused, or better implement a `copyWith` on `Parameter` and use this method in `asExpanded`, seems like it does not much more than copying the `Parameter` objects.

https://github.com/rrousselGit/freezed/blob/f6e1890453d638617e8305da9471222c90c03eda/packages/freezed/lib/src/templates/parameter_template.dart#L134-L148

Reply:
The asExpanded converts LocalParameter into Parameter, so it switches from this.property to Type property.

If you have a better name in mind, feel free to rename it! I don't quite like the name either

Reply:
About a copy vs showIsRequired, I don't really mind either of them.
Although it would be simple to do the copy

Reply:
I've now implemented a Parameter.copyWith method and removed my `toString` changes. Maybe you can take another look. 

(fwiw, i think i'm now grasping the `asExpanded` method, i didn't understand that the source parameters were of type `LocalParameter` and not `Parameter`, since it's working with a `List<Parameter>` - tbh i think i'd change this to just being a flag like `isLocalAssigned` so it wouldn't be necessary to have 3 versions of the `copyWith` method (which is currently separated into the Parameter.copyWith, `asExpanded` and `asThis` - but maybe i'm still missing something)

Reply:
The ideal scenario would be to be able to run Freezed on its own sources. But don't worry about that 😀